editor/editor_node.cpp

```diff
@@ -6121,6 +6237,24 @@ void EditorNode::reload_instances_with_path_in_edited_scenes(const String &p_ins
 
 		editor_data.restore_edited_scene_state(editor_selection, &editor_history);
 	}
+
+	scenes_modification_table.clear();
+}
+
+void EditorNode::_remove_all_not_owned_children(Node *p_node, Node *p_owner) {
+	Vector<Node *> nodes_to_remove;
```

Comment:
Not critical here, but you could `resize()` the vector to child count, to avoid appending to vector.